core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1150,11 +1150,24 @@ public void onSnapshotCompletion(Snapshot completedSnapshot, SnapshotInfo snapsh
                         @Override
                         public void onSnapshotFailure(Snapshot failedSnapshot, Exception e) {
                             if (failedSnapshot.equals(snapshot)) {
-                                logger.trace("deleted snapshot failed - deleting files", e);
+                                logger.warn("deleted snapshot failed - deleting files", e);
                                 removeListener(this);
-                                threadPool.executor(ThreadPool.Names.SNAPSHOT).execute(() ->
-                                    deleteSnapshot(failedSnapshot.getRepository(), failedSnapshot.getSnapshotId().getName(), listener, true)
-                                );
+                                threadPool.executor(ThreadPool.Names.SNAPSHOT).execute(() -> {
+                                    try {
+                                        deleteSnapshot(failedSnapshot.getRepository(),
+                                                       failedSnapshot.getSnapshotId().getName(),
+                                                       listener,
+                                                       true);
+                                    } catch (SnapshotMissingException smex) {
```

Comment:
Good point - I think we should catch a generic `SnapshotException` here, because one other possible uncaught exception would be a generic `SnapshotException` if the snapshot is incompatible.  I will make this change.